src/linear_algebra/dual.lean

```diff
@@ -455,6 +455,19 @@ begin
     rw [linear_map.map_add, h.1 _ hx, h.2 _ hy, add_zero] }
 end
 
+/-- The pullback of a submodule in the dual space along the evaluation map. -/
+def dual_annihilator_comap (Φ : submodule R (module.dual R M)) : submodule R M :=
+Φ.dual_annihilator.comap (module.dual.eval R M)
+
+lemma mem_dual_annihilator_comap_iff {Φ : submodule R (module.dual R M)} (x : M) :
+  x ∈ Φ.dual_annihilator_comap ↔ ∀ φ : Φ, φ x = 0 :=
```

Comment:
Do you think this would be more or less useful as
```suggestion
  x ∈ Φ.dual_annihilator_comap ↔ ∀ φ ∈ Φ, φ x = 0 :=
```

Reply:
I agree with you. I use `\in` for sub structures more than subtypes

Reply:
I've changed to the `\in` version but it seems that lean no longer see that `\phi x` have type `M` for some reason